lib/ansible/modules/cloud/alicloud/alicloud_vswitch_facts.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Alibaba Group Holding Limited. He Guimin <heguimin36@163.com.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+#  This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see http://www.gnu.org/licenses/.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: alicloud_vswitch_facts
+version_added: "2.4"
+short_description: Gather facts on vswitchs of Alibaba Cloud.
+description:
+     - This module fetches data from the Open API in Alicloud.
+       The module must be called from within the vswitch itself.
+
+options:
+    vpc_id:
+      description:
+        - A vpc id to list vswitches in specified vpc.
+      required: true
+      aliases: ["id"]
+    vswitch_ids:
+      required: false
+      description:
+        - A list of vswitch ids.      
```

Comment:
Please add a var  'zone_id' for the module

Reply:
Already present with alicloud_zone